sphinx/ext/extlinks.py

```diff
@@ -76,7 +76,8 @@ def check_uri(self, refnode: nodes.reference) -> None:
                 msg = __('hardcoded link %r could be replaced by an extlink '
                          '(try using %r instead)')
                 replacement = f":{alias}:`{match.groupdict().get('value')}`"
-                logger.warning(msg, uri, replacement, location=refnode)
+                logger.warning(msg, uri, replacement, location=refnode,
+                               type="extlink", subtype="hardcoded")
```

Comment:
(I'm no maintainer, just someone lurking around.)

I think you also need to update the list of warnings in the documentation of `suppress_warnings` here:

https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-suppress_warnings

Reply:
Never mind, I'm blind. Sorry.